src/software/ai/navigator/navigator.cpp

```diff
@@ -2,303 +2,325 @@
-                                 ->RobotObstacleInflationFactor()
-                                 ->value() *
-                             ROBOT_MAX_RADIUS_METERS);
-            return Obstacle(rectangle);
-        default:
-            LOG(WARNING) << "Could not convert AvoidArea " << (int)avoid_area
-                         << " to obstacle";
-    }
-
-    return std::nullopt;
+    non_path_planning_robots.insert(intent.getRobotId());
 }
 
 std::vector<std::unique_ptr<Primitive>> Navigator::getAssignedPrimitives(
     const World &world, const std::vector<std::unique_ptr<Intent>> &assignedIntents)
 {
     this->world              = world;
-    this->current_robot      = std::nullopt;
-    this->velocity_obstacles = {};
     planned_paths.clear();
+    non_path_planning_robots.clear();
+    robot_id_to_path_objectives.clear();
+    robot_id_to_move_intents.clear();
 
     auto assigned_primitives = std::vector<std::unique_ptr<Primitive>>();
     for (const auto &intent : assignedIntents)
     {
         intent->accept(*this);
-        if (this->current_robot)
+        if (current_primitive)
         {
-            if (this->current_robot->velocity().len() > 0.3)
-            {
-                this->velocity_obstacles.emplace_back(
-                    Obstacle::createVelocityObstacleWithScalingParams(
-                        this->current_robot->position(), this->current_destination,
-                        this->current_robot->velocity().len(),
-                        Util::DynamicParameters->getNavigatorConfig()
-                            ->RobotObstacleInflationFactor()
-                            ->value(),
-                        Util::DynamicParameters->getNavigatorConfig()
-                            ->VelocityObstacleInflationFactor()
-                            ->value()));
-            }
-
-            this->current_robot = std::nullopt;
+            assigned_primitives.emplace_back(std::move(current_primitive));
         }
-        assigned_primitives.emplace_back(std::move(current_primitive));
     }
 
+    Rectangle navigable_area(
+        Point(this->world.field().totalXLength(), this->world.field().totalYLength()),
+        this->world.field().totalXLength(), this->world.field().totalYLength());
+
+    auto paths = path_manager->getManagedPaths(
+        robot_id_to_path_objectives, navigable_area, getNonPathPlanningObstacles());
+    addPathsToPrimitives(paths, assigned_primitives);
+
     return assigned_primitives;
 }
 
-std::vector<Obstacle> Navigator::createCurrentObstacles(
-    const std::vector<AvoidArea> &avoid_areas, unsigned int robot_id)
+void Navigator::addPathsToPrimitives(
+    const std::map<RobotId, Path> &robot_id_to_paths,
+    std::vector<std::unique_ptr<Primitive>> &assigned_primitives)
 {
-    std::vector<Obstacle> obstacles = velocity_obstacles;
-
-    // Avoid obstacles specific to this MoveIntent
-    for (auto area : avoid_areas)
+    for (const auto &robot_id_to_path : robot_id_to_paths)
```

Comment:
nit; each element in this list isn't a map, it's a pair, suggest renaming:
`robot_id_to_paths` -> `robot_id_to_path`
`robot_id_to_path` -> `robot_id_and_path`

Reply:
done